internal/policy/root.go

```diff
@@ -111,12 +131,15 @@ func DeleteTargetsKey(rootMetadata *tuf.RootMetadata, keyID string) (*tuf.RootMe
 	if len(targetsRole.KeyIDs) <= targetsRole.Threshold {
 		return nil, ErrCannotMeetThreshold
 	}
+
 	for i, k := range targetsRole.KeyIDs {
-		if k == keyID {
-			targetsRole.KeyIDs = append(targetsRole.KeyIDs[:i], targetsRole.KeyIDs[i+1:]...)
-			break
+		if k != keyID {
```

Comment:
Could you explain the rationale here? AFAICT this isn't more efficient. Thanks!

Reply:
Also, if we're modifying this, maybe we could do something like: https://github.com/gittuf/gittuf/pull/242/files#diff-0082f62103fcad0364437363ac14e85ec846f1f3a0bd2b3c9a379c3cc4a3fccfR131-R139. This ensures that even there are duplicate instances of a key ID, we still ensure they're removed. Maybe long term, we should implement a set for these types of fields under the hood...